routing/payment_lifecycle.go

```diff
@@ -598,6 +637,15 @@ func (p *shardHandler) collectResult(attempt *channeldb.HTLCAttemptInfo) (
 			return nil, htlcswitch.ErrSwitchExiting
 		}
 
+	case <-timeoutChan:
```

Comment:
What if for some reason the HTLC is delayed along the way (a node goes down intermittently)? In this case, AFAICT, we'll stop attempting to collect this result, and fail the payment, even though it might eventually be settled as we've sent it out from the switch already.

Reply:
In that case, I don't think we would ever timeout the htlc attempt unless it has reached its CLTV. Possible related to #4363, I'll find out more about it.